data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -90,6 +95,9 @@ protected DeleteFilter(FileScanTask task, Schema tableSchema, Schema requestedSc
     this.eqDeletes = eqDeleteBuilder.build();
     this.requiredSchema = fileProjection(tableSchema, requestedSchema, posDeletes, eqDeletes);
     this.posAccessor = requiredSchema.accessorForField(MetadataColumns.ROW_POSITION.fieldId());
+
+    this.readService = ThreadPools.getWorkerPool();
```

Comment:
This should not use the worker pool. That is for job planning.

Reply:
If this is just for a default, then the default should be the existing behavior with no executor service and `CloseableIterable.concat`.